frontend/src/board/board.ts

```diff
@@ -133,4 +136,21 @@ export class Board {
 
     return legalMove 
   }
+
+
+  generateOpenPositions(boardSize: number) {
+    for (let row = 0; row < boardSize; row++) {
+      for (let col = 0; col < boardSize; col++) {
+        // if cell is not obstructed add position
+
+        if (!this.cells[row]![col]?.isObstructed && !this.cells[row]![col]?.isTarget) {
+          this.openPositions.push({row: row, column: col})
+        }
+      }
+    }
+  }
```

Comment:
What's the use-case for this?

Also I'd recommend just generating a list as you need it rather than storing a list on the board instance - since this value is computed based on the other values on the board it's likely that it will get out of sync at some point if you aren't very careful

Reply:
Okay, thanks that makes sense. I'll return a list instead and remove the open positions property.

For a use case - The idea is that it will be used for placing robot and the target chip for a random board build. After generating all the starting positions some other code could randomly assign the positions for the target chip and robots. 
.